stock_restrict_by_planned_consumed_date/models/stock_quant.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2024 Foodles (https://www.foodles.co)
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def _gather(self, *args, **kwargs):
+        quants = super()._gather(*args, **kwargs)
+        if planned_consumed_date := self.env.context.get(
+            "restrict_planned_consumed_date"
+        ):
```

Comment:
```suggestion
        planned_consumed_date = self.env.context.get("restrict_planned_consumed_date")
        if planned_consumed_date:
```

On my PR the walrus operator creates a syntax error: https://github.com/OCA/stock-logistics-workflow/actions/runs/11053805380/job/30709074146?pr=1729#step:7:1015

Reply:
Yes some runner use py3.6 (I suppose because it was the version used by odoo when the Odoo v14.0 was realeased) So maybe you want to review this base PR https://github.com/OCA/stock-logistics-workflow/pull/1724 as well :D